sorting/multi_threaded/merge_sort/multi_threaded_merge_sort.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+const threshold = 10000
+
+func parallelMergeSort(arr []int, c chan []int) {
+	if len(arr) <= 1 {
+		c <- arr
+		return
+	}
+
+	if len(arr) <= threshold {
+		timSort(arr)
+		c <- arr
+		return
+	}
+
+	middleIndex := len(arr) / 2
+	leftChan := make(chan []int)
+	rightChan := make(chan []int)
+
+	go parallelMergeSort(arr[:middleIndex], leftChan)
+	go parallelMergeSort(arr[middleIndex:], rightChan)
+
+	leftSorted := <-leftChan
+	rightSorted := <-rightChan
+
+	c <- merge(leftSorted, rightSorted)
+}
+
+func merge(leftArray, rightArray []int) []int {
+	result := make([]int, len(leftArray)+len(rightArray))
+	leftIndex, rightIndex := 0, 0
+	for i := 0; i < len(result); i++ {
+		if leftIndex >= len(leftArray) {
+			result[i] = rightArray[rightIndex]
+			rightIndex++
+			continue
+		}
+		if rightIndex >= len(rightArray) {
+			result[i] = leftArray[leftIndex]
+			leftIndex++
+			continue
+		}
+		if leftArray[leftIndex] < rightArray[rightIndex] {
+			result[i] = leftArray[leftIndex]
+			leftIndex++
+		} else {
+			result[i] = rightArray[rightIndex]
+			rightIndex++
+		}
+	}
+	return result
+}
+
+func timSort(array []int) {
+	arrayLen := len(array)
+	minRun := 32
+
+	// Sort the small runs using insertion sort
```

Comment:
It's a good practice to end comments with a `full-stop` `.` like a complete sentence. 
Good job following normal upper lower case like english!